### ISSUE TITLE
`tableStatus: CREATING` when adding a global secondary index in the table using ACK DynamoDB Controller

### ISSUE THREAD

**Issue by riddhi2910** [aws-controllers-k8s/community#1519]
**Describe the bug**
While creating the Dynamodb table using ACK Dynamodb Controller we add a global secondary index, and it is causing the table's status to be not updated, and it stays in the `CREATING` mode. 

```
Error: 2022-10-25T17:38:28.162Z	INFO	ackrt	desired resource state has changed	{"account": "xxxx", "role": "", "region": "xxx", "kind": "Table", "namespace": "preview", "name": "batchtask", "is_adopted": false, "generation": 1, "diff": [{"Path":{"Parts":["Spec","AttributeDefinitions"]},"A":[{"attributeName":"id","attributeType":"S"},{"attributeName":"batchId","attributeType":"S"},{"attributeName":"status","attributeType":"S"}],"B":[{"attributeName":"batchId","attributeType":"S"},{"attributeName":"id","attributeType":"S"},{"attributeName":"status","attributeType":"S"}]}]}
2022-10-25T17:38:28.162Z	INFO	ackrt	updated resource	{"account": "xxxx", "role": "", "region": "xxxx", "kind": "Table", "namespace": "xxx", "name": "batchtask", "is_adopted": false, "generation": 1}
```

**Steps to reproduce**

Table:

```yaml
apiVersion: dynamodb.services.k8s.aws/v1alpha1
kind: Table
metadata:
  creationTimestamp: "2022-10-25T17:09:08Z"
  finalizers:
  - finalizers.dynamodb.services.k8s.aws/Table
  generation: 1
  labels:
    app: xxxxx
  name: batchtask
  namespace: xxxxxx
spec:
  attributeDefinitions:
  - attributeName: id
    attributeType: S
  - attributeName: batchId
    attributeType: S
  - attributeName: status
    attributeType: S
  globalSecondaryIndexes:
  - indexName: BatchIdStatusIndex
    keySchema:
    - attributeName: batchId
      keyType: HASH
    - attributeName: status
      keyType: RANGE
    projection:
      projectionType: ALL
    provisionedThroughput:
      readCapacityUnits: 25
      writeCapacityUnits: 25
  keySchema:
  - attributeName: id
    keyType: HASH
  provisionedThroughput:
    readCapacityUnits: 25
    writeCapacityUnits: 25
  tableName: batchtask
  tags:
  - key: xxxx
    value: xxxx
  - key: zzzz
    value: zzzz
```

**Expected outcome**
A concise description of what you expected to happen.

**Environment**

* Kubernetes version 1.22
* Using EKS (yes/no), if so version?  EKS Yes
* AWS service targeted (S3, RDS, etc.) Dynamodb

**Comment by riddhi2910** [comment]
@jaypipes

**Comment by jaypipes** [comment]
@riddhi2910 apologies for the super-long delay on this. Working on it now.

**Comment by A-Hilaly** [comment]
Fixed and released in dynamodb `v1.1.0`
/close
